opentelemetry-exporter-gcp-trace/lib/opentelemetry/exporter/google_cloud_trace/span_exporter.rb

```diff
@@ -0,0 +1,109 @@
+# Copyright 2023 Google LLC
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     https://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# frozen_string_literal: true
+
+require "google/cloud" unless defined? Google::Cloud.new
+require "google/cloud/config"
```

Comment:
It would be better if we can avoid depending on google-cloud-core. That's used by the handwritten clients, and we're not using the trace handwritten client here. For the `default_project_id` method, go ahead and use `Google::Cloud.env` and feel free to depend on the `google-cloud-env` gem (which we will probably have to because we'll need the resource detection) but don't use `Google::Cloud.configure`.